Example/NYTViewController.m

```diff
@@ -23,101 +24,67 @@ typedef NS_ENUM(NSUInteger, NYTViewControllerPhotoIndex) {
 @interface NYTViewController () <NYTPhotosViewControllerDelegate>
 
 @property (weak, nonatomic) IBOutlet UIButton *imageButton;
-@property (nonatomic) NSArray *photos;
+@property (nonatomic) NYTPhotoViewerArrayDataSource *dataSource;
 
 @end
 
 @implementation NYTViewController
 
 - (IBAction)imageButtonTapped:(id)sender {
-    self.photos = [[self class] newTestPhotos];
-    
-    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithPhotos:self.photos initialPhoto:nil delegate:self];
+    self.dataSource = [self.class newTimesBuildingDataSource];
+
+    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithDataSource:self.dataSource initialPhoto:nil delegate:self];
 
     [self presentViewController:photosViewController animated:YES completion:nil];
     
-    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithPhotos:self.photos];
+    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
+//    [self switchDataSourceOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
```

Comment:
Perhaps provide a little explanation here. Is you intention that only one of these two lines be uncommented or should we simply uncomment line 41 to see the data source switch after a delay. It works both ways, but it would be nice to know your intention. Or maybe use an if statement to make it more clear:

```
BOOL switchDataSource = false
[self updateImagesOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
if (switchDataSource) {
    [self switchDataSourceOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
}
```